Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -1117,6 +1117,24 @@ class ConstantFoldingPhase : public Phase {
 
                 break;
             }
+
+            case HasStructureWithFlags: {
+                const AbstractValue& abstractValue = m_state.forNode(node->child1());
+                unsigned flags = node->structureFlags();
+                ASSERT(flags);
+
+                if (abstractValue.m_type && !(abstractValue.m_type & ~SpecCell) && abstractValue.m_structure.isFinite()) {
+                    dataLogLn("CF");
+                    bool anyStructureHasAnyFlag = false;
+                    abstractValue.m_structure.forEach([&] (RegisteredStructure structure) {
+                        anyStructureHasAnyFlag |= structure->hasAnyOfBitFieldFlags(flags);
+                    });
+                    m_graph.convertToConstant(node, jsBoolean(anyStructureHasAnyFlag));
```

Comment:
Converting it to `jsBoolean(anyStructureHasAnyFlag)` looks wrong, right?
If one structure does not have this flag, but if the other candidate structures have, then it becomes true, and this is not correct.
You can fold to a constant only when all the structure agrees one result.

Reply:
Also, you need to `change = true` when you changed the graph.

Reply:
Yeah my bad, changed, thanks!